designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+
+            _frameworkProfile = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            Platform = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            PlatformVersion = NormalizeVersion(platformVersion ?? FrameworkConstants.EmptyVersion);
+        }
+```
+
+### GetDotNetFrameworkName
+
+This method needs to be updated to append the Platform and PlatformVersion. I believe the format is supposed to be:
+
+```csharp
+string.Format(CultureInfo.InvariantCulture, "Platform={0},PlatformVersion={0}", framework.Platform, framework.PlatformVersion);
+```
+
+### GetShortFolderName
+
+This method needs to be updated such that when `IsNet5Era` is `true` and `Platform` is not `null`, `"-{Platform}[PlatformVersion]` is appended to the `net5.0` short tfm. This should be similar to how profiles are appended.
+
+## class NuGetFrameworkFactory
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs
+
+### ParseFrameworkName
+
+This seems to be the core of the "long" tfm parsing logic. This needs to be modified to detect `Platform=` and `PlatformVersion=` and extract those accordingly. It should also skip parsing out `Profile` if it detects that the framework it's currently parsing is net5.0+. If an invalid Platform is detected, it should error out.
+
+### ParseFolder
+
+This seems to be the core of the "short" tfm parsing logic. As with `ParseFrameworkName`, this needs to be modified such that, when the identifier is net5.0+, it will check `FrameworkConstants.FrameworkPlatforms` and verify that the given platform is one of the accepted ones (per the net5.0 spec linked at top). If an invalid Platform is detected, it should error out.
+
+### RawParse
+
+This... mostly seems to do what I think it needs to do? It should probably have a big fat comment noting that we're piggybacking off profile parsing (keeping the net5.0+ dependent logic in `ParseFolder`). Maybe renaming variables to `profileOrPlatform` for clarity.
+
+## class FrameworkConstants
+
+Source: src/NuGet.Core/NuGet.Frameworks/FrameworkConstants.cs
+
+### (new!) FrameworkPlatforms
+
+A new `internal static HashSet<string> FrameworkPlatforms` field should be added and populated with all valid platforms (`android`, `ios`, `macos`, `tvos`, `watchos`, `windows`). As of this writing, it's **unclear** what to do about Tizen and Unity, and whether they'll be regular Platforms or a TFI or whatever.
+
+## class CompatibilityProvider
+
+Source: src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs
+
+### IsCompatibleWithTargetCore
+
+This method needs to be modified such that if there's a Platform present in the `candidate`, the platform name has to match. Additionally, an `IsVersionCompatible(target.PlatformVersion, candidate.PlatformVersion)` should be made if both have a `PlatformVersion` included. This will need to check for null `PlatformVersion`s, as well.

```

Comment:
Mention of `null` versions, but my other comment is on logic that refers to 0.0 as a missing Version.

Reply:
oh thank you. I needed to update this after we changed to 0.0 as a sentinel :)